libs/sdk-core/src/swap_in/swap.rs

```diff
@@ -181,13 +181,38 @@ impl BTCReceiveSwap {
             .get_node_state()?
             .ok_or(anyhow!("Node info not found"))?;
 
+        // Calculate max_allowed_deposit based on absolute max and current node state
+        let fn_max_allowed_deposit = |max_allowed_deposit_abs: i64| {
+            std::cmp::min(
+                (node_state.max_receivable_msat / 1000) as i64,
+                max_allowed_deposit_abs,
+            )
+        };
+
         // check first that we don't already have an unused swap
         if let Some(unused_swap) = self.list_unused()?.first().cloned() {
             info!("Found unused swap when trying to create new swap address");
+            let bitcoin_address = unused_swap.bitcoin_address.clone();
+
+            // Check max_allowed_deposit and, if it changed, persist and validate changes
+            let current_max = fn_max_allowed_deposit(unused_swap.max_allowed_deposit_abs);
+            let res_swap = match current_max == unused_swap.max_allowed_deposit {
+                true => unused_swap,
+                false => {
+                    info!("max_allowed_deposit for this swap has changed, updating it");
+                    let mut new_swap = unused_swap.clone();
+
+                    new_swap.max_allowed_deposit = current_max;
+                    new_swap.validate_max_allowed_deposit()?;
```

Comment:
If we are adding validation here perhaps it is a good idea to add a specific error? `UnsupportedSwapLimits` ?

Reply:
Done: a29023c2e97b9c13a86c04944e04a7b021ee6eea